mewarpx/mewarpx/sim_control.py

```diff
@@ -93,3 +98,26 @@ def check_criteria(self):
             logger.info(terminate_statement)
 
         return all_good
+
+    def write_results(self):
+        """Create results.txt file, and write to it if write_func is set.
+        The file signifies that the simulation ran to completion."""
+        results_string = ""
+        if callable(self._write_func):
+            results_string = self._write_func()
+
+        if mwxrun.me == 0:
+            mwxutil.mkdir_p(WarpXDiagnostic.DIAG_DIR)
+            with open(os.path.join(WarpXDiagnostic.DIAG_DIR, "results.txt"), 'a') as results_file:
+                results_file.write(results_string)
+
+    def set_write_func(self, func):
```

Comment:
In the way you are using this it might be best to make it a `setter` see https://www.geeksforgeeks.org/getter-and-setter-in-python/

Reply:
I originally didn't do this since it would add several lines of unused code (the getter), but I can change it if you think it looks better.

Reply:
I don't have a strong preference but it does open the possibility to set the write_func directly without going through the check that it is a callable. Can you not make a setter without explicitly declaring the getter?

Reply:
I believe so, since you would need a `@property` block like 
```
@property
def write_func(self):
    return self.write_func
```

Reply:
I think I'm with @KZhu-ME on this one, it would require a getter so this seems fine. Prefacing `_write_func` with the underscore is the pythonic way of saying 'If you set this directly and screw stuff up it's your own fault', so I don't see an issue there.